lua/ttt2/libraries/vguihandler.lua

```diff
@@ -0,0 +1,236 @@
+---
+-- A vgui handler
+-- @author Mineotopia
+
+AddCSLuaFile()
+
+-- the rest of the draw library is client only
+if SERVER then return end
+
+vguihandler = vguihandler or {
+	menuCache = {
+		frame = nil,
+		hidden = nil
```

Comment:
Why do you need both fields? Afaik only one is used anyways, and the other one is always nil? Wouldn't it be more fail-safe to use a boolean to express this state?